packages/variants/src/variants.ts

```diff
@@ -6,13 +6,13 @@ function makeThemeFromPartials(
   partialPalette: Partial<IPalette>,
   partialSemantic: Partial<ISemanticColors>
 ): ITheme {
-  return createTheme({
-    ...originalTheme,
-    ...{
-      palette: { ...originalTheme.palette, ...partialPalette },
-      semanticColors: { ...originalTheme.semanticColors, ...partialSemantic }
-    }
-  });
+  // Create variant palette
+  let variantTheme = createTheme({ palette: { ...originalTheme.palette, ...partialPalette } });
+  // Change semantic colors to use updated variant palette values
+  variantTheme.semanticColors = { ...variantTheme.semanticColors, ...partialSemantic };
+  // Fill in the rest of the theme
+  variantTheme = { ...originalTheme, ...variantTheme, semanticColors: variantTheme.semanticColors };
```

Comment:
this should be:
variantTheme = { ...originalTheme, palette: ...variantTheme.palette, semanticColors: variantTheme.semanticColors };

Reply:
@phkuo I can't seem to get this working. I'm getting a linting error on `variantTheme=`:

```
Type '{ semanticColors: ISemanticColors; themeDarker: string; themeDark: string; themeDarkAlt: string; themePrimary: string; themeSecondary: string; themeTertiary: string; themeLight: string; ... 46 more ...; typography?: IPartialTypography | undefined; }' is not assignable to type 'ITheme'.

 Object literal may only specify known properties, and 'themeDarker' does not exist in type 'ITheme'.
```

Any ideas on what should be different?

Reply:
oh, it should be this:
`variantTheme = { ...originalTheme, palette: variantTheme.palette, semanticColors: variantTheme.semanticColors };`
without the ... before variantTheme.palette

Reply:
@phkuo  Ok, now I have a new linting error.
```
Type '{ palette: IPalette; semanticColors: ISemanticColors; fonts?: Partial<IFontStyles> | undefined; isInverted?: boolean | undefined; disableGlobalClassNames?: boolean | undefined; typography?: IPartialTypography | undefined; }' is not assignable to type 'ITheme'.
  Types of property 'fonts' are incompatible.
    Type 'Partial<IFontStyles> | undefined' is not assignable to type 'IFontStyles'.
      Type 'undefined' is not assignable to type 'IFontStyles'.
```
Looking into it now. Any idea where the undefined thing is coming from?

Reply:
it means it's an optional
okay i see, it's because the originalTheme we're taking in is an IPartialTheme instead of an ITheme
okay so what you'll want to do is in the getXVariant() is change the makeThemeFromPartials() call to take `fullTheme` instead of `theme`, it's already defined in each getXVariant() function
and then change the type of originalTheme: IPartialTheme here to ITheme